interfaces/builtin/README.md

```diff
@@ -333,3 +344,240 @@ a store assertion):
+
+For connections involving the snap, no rules are checked and it is always
+allowed.
+
+For auto-connection the base-declaration and any rule for the other side if
+available is considered, this usually means no auto-connection will happen
+unless allowed by the base-declaration or by a slot-side snap-declaration.
+
+## Base declaration policy patterns
+
+This section discusses the patterns to follow when writing base declaration
+rules for an interface (via setting
+`commonInterface.baseDeclarationSlots/Plugs`) depending on the security
+characteristics of the interface.
+
+Slots for a specific interface can be provided by the system snap (so called
+implicit slots) or by an application snap or sometimes both.
+
+The base declaration rules are for almost all interfaces and whenever possible
+written as slot-side-only rules. As reminder, importantly, items are not merged
+between the slots and plugs or between the base declaration and snap
+declaration for a particular type of rule. This means if one specifies a
+connection rule both on the slot-side and plug-side in the base declaration,
+only the plug side is used (plug-side override the slot side). Installation
+rules target their side only, a slot-side installation rule is about allowing
+snaps with a slot of the given interface, same for a plug-side installation
+rule being about snaps with a plug of the given interface.
+
+Interfaces can be broadly categorized as:
+
+- auto-connected implicit slots (eg, network)
+- manually connected implicit slots (eg, bluetooth-control)
+- auto-connected app-provided slots  (eg, mir)
+- manually connected app-provided slots (eg, bluez)
+
+such that they will follow this pattern:
+
+    slots:
+      auto-connected-implicit-slot:
+        allow-installation:
+          slot-snap-type:
+            - core                     # implicit slot
+        #allow-auto-connection: true   # allow auto-connect, default
+  
+      manual-connected-implicit-slot:
+        allow-installation:
+          slot-snap-type:
+            - core                     # implicit slot
+        deny-auto-connection: true     # force manual connect
+  
+      auto-connected-provided-slot:
+        allow-installation:
+          slot-snap-type:
+            - app                      # app provided slot
+        deny-connection: true          # require allow-connection in snap decl
+  
+      manual-connected-provided-slot:
+        allow-installation:
+          slot-snap-type:
+            - app                      # app provided slot
+        deny-connection: true          # require allow-connection in snap decl
+        deny-auto-connection: true     # force manual connect
+
+As some slots can be provided by both the system or an application snap, some interfaces will be in more than one category.
+
+Auto-connection should be allowed in the base-declaration if the access to the
+resources provided by the interface does not have security implications.
```

Comment:
```suggestion
Auto-connection should be allowed in the base-declaration if the access to the
resources provided by the interface do not have security implications.
```

Reply:
confused about this change,  "the access" ... "does", no?